packages/platform-android/src/commands/runAndroid/runOnAllDevices.js

```diff
@@ -7,103 +7,85 @@
 }
 
+function toPascalCase(value: string) {
+  return value[0].toUpperCase() + value.slice(1);
+}
+
 function runOnAllDevices(
-  args: Object,
+  args: FlagsT,
   cmd: string,
   packageNameWithSuffix: string,
   packageName: string,
   adbPath: string,
 ) {
   try {
-    const gradleArgs = [];
-
-    if (args.installDebug) {
-      gradleArgs.push(getCommand(args.appFolder, args.installDebug));
-    } else if (args.variant) {
-      gradleArgs.push(
-        `${getCommand(
-          args.appFolder,
-          'install',
-        )}${args.variant[0].toUpperCase()}${args.variant.slice(1)}`,
-      );
-    } else if (args.flavor) {
-      logger.warn('--flavor has been deprecated. Use --variant instead');
-      gradleArgs.push(
-        `${getCommand(
-          args.appFolder,
-          'install',
-        )}${args.flavor[0].toUpperCase()}${args.flavor.slice(1)}`,
-      );
-    } else {
-      gradleArgs.push(getCommand(args.appFolder, 'installDebug'));
-    }
+    const gradleArgs = [
+      args.task
+        ? getTaskName(args.appFolder, args.task)
+        : getTaskName(args.appFolder, 'install' + toPascalCase(args.variant)),
+    ];
 
-    logger.info(
-      `Building and installing the app on the device (cd android && ${cmd} ${gradleArgs.join(
-        ' ',
-      )})...`,
+    logger.info('Installing the app...');
+    logger.debug(
+      `Running command "cd android && ${cmd} ${gradleArgs.join(' ')}"`,
     );
 
-    execFileSync(cmd, gradleArgs, {
-      stdio: [process.stdin, process.stdout, process.stderr],
-    });
-  } catch (e) {
-    logger.error(
-      'Could not install the app on the device, read the error above for details.\n' +
-        'Make sure you have an Android emulator running or a device connected and have\n' +
-        'set up your Android development environment:\n' +
-        'https://facebook.github.io/react-native/docs/getting-started.html',
-    );
-    // stderr is automatically piped from the gradle process, so the user
-    // should see the error already, there is no need to do
-    // `logger.info(e.stderr)`
-    return Promise.reject(e);
+    execFileSync(cmd, gradleArgs, {stdio: ['inherit', 'inherit', 'pipe']});
+  } catch (error) {
+    throw createInstallError(error);
   }
   const devices = adb.getDevices(adbPath);
-  if (devices && devices.length > 0) {
-    devices.forEach(device => {
-      tryRunAdbReverse(args.port, device);
-      tryLaunchAppOnDevice(
-        device,
-        packageNameWithSuffix,
-        packageName,
-        adbPath,
-        args.mainActivity,
-      );
-    });
-  } else {
-    try {
-      // If we cannot execute based on adb devices output, fall back to
-      // shell am start
-      const fallbackAdbArgs = [
-        'shell',
-        'am',
-        'start',
-        '-n',
-        `${packageNameWithSuffix}/${packageName}.MainActivity`,
-      ];
-      logger.info(
-        `Starting the app (${adbPath} ${fallbackAdbArgs.join(' ')}...`,
```

Comment:
This else branch is gone, right?

Reply:
yep, the logic was quite duplicated here